fiftyone/utils/iou.py

```
@@ -650,52 +792,85 @@
 
-def _compute_keypoint_similarities(preds, gts, classwise=False):
-    sims = np.zeros((len(preds), len(gts)))
-    for j, gt in enumerate(gts):
-        for i, pred in enumerate(preds):
+def _compute_keypoint_similarities(preds, gts, classwise=False, sparse=False):
+    is_symmetric = preds is gts
+
+    if sparse:
+        sims = defaultdict(list)
+    else:
+        sims = np.zeros((len(preds), len(gts)))
```

Ruff

Use ternary operator `sims = defaultdict(list) if sparse else np.zeros((len(preds), len(gts)))` instead of `if`-`else`-block

Replace `if`-`else`-block with `sims = defaultdict(list) if sparse else np.zeros((len(preds), len(gts)))`

(SIM108)